implementations/boon/src/main.rs

```diff
@@ -0,0 +1,39 @@
+use std::{error::Error, fs::File, io::{BufReader, BufRead}, time::Instant};
+use boon::{Compiler, Schemas};
+use serde_json::Value;
+use std::env;
+
+fn main() -> Result<(), Box<dyn Error>> {
+  // Get arguments
+  let args: Vec<String> = env::args().collect();
+  let example_folder = &args[1];
+
+  // Get the schema and instance paths
+  let schema_file =   std::fs::canonicalize(example_folder.to_owned() + "/schema.json")?;
+  let instance_file = std::fs::canonicalize(example_folder.to_owned() + "/instances.jsonl")?;
+
+  // Read the instance file
+  let file = File::open(&instance_file)?;
+  let reader = BufReader::new(file);
+
+  // Compile the schema
+  let mut schemas = Schemas::new();
+  let mut compiler = Compiler::new();
+  let sch_index = compiler.compile(schema_file.to_str().ok_or("NULL")?, &mut schemas)?;
+
+  // Validate each instance
+  let start = Instant::now();
```

Comment:
One improvement you should do there, is that you are doing I/O on the JSONL file, including parsing each line with Serde, on the code block that is measuring the benchmark, which can affect results.

You likely should:

- First parse the JSONL and each of its entries
- Store them in an array/vector or whatever Rust uses
- THEN start measuring, loop on the parsed list, and validate

Reply:
Damn! I let that! Thanks ❤️